ambry-server/src/main/java/com.github.ambry.server/AmbryRequests.java

```diff
@@ -992,17 +992,7 @@ private ServerErrorCode validateRequest(PartitionId partition, RequestOrResponse
       return ServerErrorCode.Bad_Request;
     }
     if (!skipPartitionAndDiskAvailableCheck) {
```

Comment:
Should this `if (!skipPartitionAndDiskAvailableCheck)` include ` 2. ensure the disk for the partition/replica is available` only?

Reply:
`skipPartitionAndDiskAvailableCheck` is used when starting a specific store. Since the request is validated at the very beginning when the store is not started (partition not available), it is necessary to skip this check and proceed rather than deny the request.